rosbridge_server/scripts/rosbridge_websocket.py

```diff
@@ -55,184 +60,240 @@
-        idx = sys.argv.index("--delay_between_messages") + 1
-        if idx < len(sys.argv):
-            RosbridgeWebSocket.delay_between_messages = float(sys.argv[idx])
-        else:
-            print("--delay_between_messages argument provided without a value.")
-            sys.exit(-1)
-
-    if "--max_message_size" in sys.argv:
-        idx = sys.argv.index("--max_message_size") + 1
-        if idx < len(sys.argv):
-            value = sys.argv[idx]
-            if value == "None":
-                RosbridgeWebSocket.max_message_size = None
+
+class RosbridgeWebsocketNode(Node):
+    def __init__(self):
+        super().__init__('rosbridge_websocket')
+
+        ##################################################
+        # Parameter handling                             #
+        ##################################################
+        retry_startup_delay = self.get_parameter('retry_startup_delay').value
+        retry_startup_delay = 2.0 if retry_startup_delay is None
+                                  else retry_startup_delay  # seconds
+
+
+        use_compression = self.get_parameter('use_compression').value
+        RosbridgeWebSocket.use_compression = False if use_compression is None
+                                                   else use_compression
+
+        # get RosbridgeProtocol parameters
+        fragment_timeout = self.get_parameter('fragment_timeout').value
+        RosbridgeWebSocket.fragment_timeout = RosbridgeWebSocket.fragment_timeout
+                                              if fragment_timeout is None
+                                              else fragment_timeout
+
+        delay_between_messages = self.get_parameter('delay_between_messages').value
+        RosbridgeWebSocket.delay_between_messages = RosbridgeWebSocket.delay_between_messages 
+                                                    if delay_between_messages is None
+                                                    else delay_between_messages
+
+        max_message_size = self.get_parameter('max_message_size').value
+        RosbridgeWebSocket.max_message_size = RosbridgeWebSocket.max_message_size
+                                              if max_message_size is None
+                                              else max_message_size
+
+        unregister_timeout = self.get_parameter('unregister_timeout').value
+        RosbridgeWebSocket.unregister_timeout = RosbridgeWebSocket.unregister_timeout
+                                                if unregister_timeout is None
+                                                else unregister_timeout
+
+        bson_only_mode = self.get_parameter('retry_startup_delay').value
+        bson_only_mode = False if bson_only_mode is None
+                               else bson_only_mode
+
+        if RosbridgeWebSocket.max_message_size == "None":
+            RosbridgeWebSocket.max_message_size = None
+
+        # SSL options
+        certfile = self.get_parameter('certfile').value
```

Comment:
I think this can raise an error if you don't allow undeclared parameters. I would allow them, or use `get_paramer_or` again here.
I think you will need `automatically_declare_parameter_overrides` too, right?

Reply:
Changed to `declare_parameter` in https://github.com/RobotWebTools/rosbridge_suite/pull/419/commits/9161521da0b402b5fa6720e54cb61bd156ec58f6.